src/backends/postgresql/session.cpp

```diff
@@ -29,6 +30,126 @@ void hard_exec(postgresql_session_backend & session_backend,
     postgresql_result(session_backend, PQexec(conn, query)).check_for_errors(errMsg);
 }
 
+// helper function to quote a string before sending to PostgreSQL
+std::string quote(PGconn * conn, std::string& s)
+{
+    int error_code;
+    char *retv = new char[2 * s.length() + 1];
+    size_t len_esc = PQescapeStringConn(conn, retv, s.c_str(), s.length(), &error_code);
```

Comment:
Sorry, when I recommended using `std::string` I meant to do it _instead_ of manually allocated buffer, not in addition to it. I.e. could we please replace this with

```suggestion
    std::string retv;
    retv.resize(2 * s.length() + 3);
    retv[0] = '\'';
    size_t len_esc = PQescapeStringConn(conn, retv.data() + 1, s.c_str(), s.length(), &error_code);
```

and remove `delete []` below?

Reply:
Thank you - I was too much caught up in bit keeping... (I had to const_cast retv.data() but it seems good otherwise)